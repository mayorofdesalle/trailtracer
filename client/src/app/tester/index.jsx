import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
