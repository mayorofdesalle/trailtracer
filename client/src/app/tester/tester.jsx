import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };