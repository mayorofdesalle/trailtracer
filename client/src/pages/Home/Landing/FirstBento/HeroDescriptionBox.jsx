import { useTranslation } from 'react-i18next';
import styled, { useTheme } from 'styled-components';

import { slideFromBottom } from '@components/style/anims';
import { pseudoBentoArrowTop } from '@components/ui/Bento/Arrows';
import BentoBox from '@components/ui/Bento/BentoBox';
import Text from '@components/ui/Text';

const HeroDescriptionBoxInner = styled(BentoBox)`
    grid-area: HeroDescriptionBox;
    --arrowColor: ${({ theme }) => theme.colors.primary};
    ${pseudoBentoArrowTop}

    &::before {
        animation: ${slideFromBottom} 1s ease-in-out;
    }
`;

/**
 * HeroDescriptionBox
 * @description
 * Styled bento box to contain the hero description.
 **/
const HeroDescriptionBox = () => {
    const theme = useTheme();
    const { t } = useTranslation();

    return (
        <HeroDescriptionBoxInner $glass>
            <Text $heading $color={theme.colors.text}>
                {t('landingPage.title')}
            </Text>
        </HeroDescriptionBoxInner>
    );
};

export default HeroDescriptionBox;