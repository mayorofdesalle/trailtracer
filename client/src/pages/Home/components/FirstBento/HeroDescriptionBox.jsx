import styled from 'styled-components';

import BentoBox from '@components/ui/BentoBox';
import { glassMorph } from '@components/misc/mixins';
import { rotateInFB, fadeIn } from '@components/misc/anims';

import { pseudoBentoArrowTop } from '../mixins';

/**
 * HeroDescriptionBox
 * @description
 * This is a styled bento box that is used to contain the hero description.
 **/
const HeroDescriptionBox = styled(BentoBox)`
    outline: 2px solid ${({ theme }) => theme.colors.primaryTransparent};
    animation: ${rotateInFB} 1s ease-in-out;

    --arrowColor: ${({ theme }) => theme.colors.primary};
    ${pseudoBentoArrowTop}

    ${glassMorph}

    & > * {
        animation: ${fadeIn} 1s ease-in-out;
        line-height: 1.43;
    }
`;

export default HeroDescriptionBox;