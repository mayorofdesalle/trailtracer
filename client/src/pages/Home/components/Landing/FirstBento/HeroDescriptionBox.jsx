import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/Box';
import { pseudoBentoArrowTop } from '@components/misc/mixins';
import { slideFromBottom } from '@components/misc/anims';

/**
 * HeroDescriptionBox
 * @description
 * This is a styled bento box that is used to contain the hero description.
 **/
const HeroDescriptionBox = styled(BentoBox)`
    grid-area: HeroDescriptionBox;
    --arrowColor: ${({ theme }) => theme.colors.primary};
    ${pseudoBentoArrowTop}

    &::before {
        animation: ${slideFromBottom} 1s ease-in-out;
    }
`;

export default HeroDescriptionBox;