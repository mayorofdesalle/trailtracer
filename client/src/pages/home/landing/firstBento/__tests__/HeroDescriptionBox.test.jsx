import { describe, expect, it, render, screen } from '@app/tester';

import HeroDescriptionBox from '../HeroDescriptionBox';

describe('HeroDescriptionBox', () => {

    it('should render correctly', () => {
        render(<HeroDescriptionBox data-testid='hero-description-box' />);
        const heroDescriptionBox = screen.getByTestId('hero-description-box');

        expect(heroDescriptionBox).toBeInTheDocument();
        expect(heroDescriptionBox).toHaveStyle('grid-area: HeroDescriptionBox');
        expect(heroDescriptionBox).toHaveTextContent(/title/i);
    });

});