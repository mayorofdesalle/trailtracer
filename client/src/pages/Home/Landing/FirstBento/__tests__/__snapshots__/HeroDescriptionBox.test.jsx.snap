// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeroDescriptionBox > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-irLvIq kHwwsn dHdhmG kOBRmR"
  >
    <div
      class="sc-iBdnpw eFPdJJ"
    >
      landingPage.title
    </div>
  </div>
</DocumentFragment>
`;
